proto/substrait/extended_expression.proto

```diff
@@ -19,33 +19,42 @@ message ExpressionReference {
     AggregateFunction measure = 2;
   }
   // Field names in depth-first order
+  //
+  // For primitive types, this will just be a single name. Nested types may
+  // have multiple names.
   repeated string output_names = 3;
```

Comment:
Is this a required field?  The description below seems to suggest it is but it might not hurt to mention that here.